src/MessageRequest.jl

```diff
@@ -1,102 +1,23 @@
 module MessageRequest
 
-export body_is_a_stream, body_was_streamed, setuseragent!, resource
+export setuseragent!
 
-
-import ..Layer, ..request
-using ..IOExtras
 using URIs
-using ..Messages
-import ..Messages: bodylength
 import ..Headers
-import ..Form, ..content_type
+using ..Messages
 
-"""
-"request-target" per https://tools.ietf.org/html/rfc7230#section-5.3
-"""
-resource(uri::URI) = string( isempty(uri.path)     ? "/" :     uri.path,
-                            !isempty(uri.query)    ? "?" : "", uri.query,
-                            !isempty(uri.fragment) ? "#" : "", uri.fragment)
+export messagelayer
 
 """
-    request(MessageLayer, method, ::URI, headers, body) -> HTTP.Response
+    messagelayer(method, ::URI, headers, body) -> HTTP.Response
```

Comment:
We now require `messagelayer` to be the 1st/outermost layer in our stack as it translates the method, url, headers, and body into the `Request` object that all other layers consume.